### PR TITLE
Contact

### PR DESCRIPTION
**Pull request by TheYisusGithub**

### CODE REVIEW

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * Jala Foundation, ("Confidential Information").  You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Jala Foundation.
+ */
+package com.foundation.salesforce.core.api;
+
+import com.foundation.salesforce.core.restClient.Authentication;
+import com.foundation.salesforce.core.restClient.RestClientApi;
+import io.restassured.response.Response;
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/core/api/ContactApi.java:18]: Please, place a separator blank line.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+ * accordance with the terms of the license agreement you entered into
+ * with Jala Foundation.
+ */
+package com.foundation.salesforce.steps;
+
+import com.foundation.salesforce.core.api.ContactApi;
+import com.foundation.salesforce.core.restClient.RestClientApi;
+import com.foundation.salesforce.core.utils.EndPoints;
+import com.foundation.salesforce.entities.Contact;
+import cucumber.api.java.en.Given;
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place separator blank lines between groups of libraries.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    @Then("User get a {string} status code with firstName, email")
+    public void userGetAStatusCodeWithFirstNameEmail(String statusCode) {
+        Assert.assertEquals(response.getStatusCode(), Integer.parseInt(statusCode));
+    }
+
+    /**
+     * Method to let user specifies new contact with first name, email and title.
+     *
+     * @param inputContent
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for the parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    @When("User send de request post to contact endpoint with firstName, email")
+    public void userSendDeRequestPostToContactEndpointWithFirstNameEmail() {
+        response = RestClientApi.getInstance().post(EndPoints.CONTACT_ENDPOINT);
+    }
+
+    /**
+     * Method to led me get a status code with first name and email.
+     *
+     * @param statusCode
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    @Then("User get a {string} status code with firstName")
+    public void userGetAStatusCodeWithFirstName(String statusCode) {
+        Assert.assertEquals(response.getStatusCode(), Integer.parseInt(statusCode));
+    }
+
+    /**
+     * Method to let me specifies new contact with the first name and email.
+     *
+     * @param inputContent
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    @When("User send de request post to contact endpoint with firstName")
+    public void userSendDeRequestPostToContactEndpointWithFirstName() {
+        response = RestClientApi.getInstance().post(EndPoints.CONTACT_ENDPOINT);
+    }
+
+    /**
+     * Method let me get the status code with first name.
+     *
+     * @param statusCode
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    @Then("User get a {string} status code")
+    public void userGetAStatusCode(String statusCode) {
+        Assert.assertEquals(response.getStatusCode(), Integer.parseInt(statusCode));
+    }
+
+    /**
+     * Method to led a specifies a new contact.
+     *
+     * @param inputContent
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    @When("User send de request post to contact endpoint")
+    public void userSendDeRequestPostToContactEndpoint() {
+        response = RestClientApi.getInstance().post(EndPoints.CONTACT_ENDPOINT);
+    }
+
+    /**
+     * Method to let compare a recuest.
+     *
+     * @param statusCode
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    @Given("a user sing in the web page SalesForce")
+    public void autheticationUser() {
+        contactApi.getInstance();
+    }
+
+    /**
+     * Method to let a build a new contact.
+     *
+     * @param inputContent
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for input parameters.

File: src/main/java/com/foundation/salesforce/entities/Contact.java [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+     * @return Title.
+     */
+    public String getTitle() {
+        return Title;
+    }
+
+    /**
+     * Method to let a title to the user.
+     *
+     * @param title
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/entities/Contact.java]: Please, place descriptions for input parameters.

File: src/main/java/com/foundation/salesforce/entities/Contact.java [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+     * @param email
+     */
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    /**
+     * Method to let return the title to the user.
+     *
+     * @return Title.
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/entities/Contact.java]: Please, place descriptions for input parameters.

File: src/main/java/com/foundation/salesforce/entities/Contact.java [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+     * @return
+     */
+    public String getEmail() {
+        return email;
+    }
+
+    /**
+     * Method to let get a email from the user.
+     *
+     * @param email
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/entities/Contact.java]: Please, place descriptions for input parameters.

File: src/main/java/com/foundation/salesforce/entities/Contact.java [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+     * @param firstName
+     */
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+
+    /**
+     * Method to let get the email form the user.
+     *
+     * @return
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/entities/Contact.java]: Please, place descriptions for method returns.

File: src/main/java/com/foundation/salesforce/entities/Contact.java [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+     * @return firstName
+     */
+    public String getFirstName() {
+        return firstName;
+    }
+
+    /**
+     * Method to let the firstName to the user.
+     *
+     * @param firstName
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/entities/Contact.java]: Please, place descriptions for input parameters.

File: src/main/java/com/foundation/salesforce/entities/Contact.java [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+     * @param lastName
+     */
+    public void setLastName(String lastName) {
+        this.lastName = lastName;
+    }
+
+    /**
+     * Method to let the return the firstName from the first name.
+     *
+     * @return firstName
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/entities/Contact.java]: Please, place descriptions for returns.

File: src/main/java/com/foundation/salesforce/entities/Contact.java [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+     * @return lastName
+     */
+    public String getLastName() {
+        return lastName;
+    }
+
+    /**
+     * Method to let lastName from the user.
+     *
+     * @param lastName
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/entities/Contact.java]: Please, place descriptions for input parameters.

File: src/main/java/com/foundation/salesforce/entities/Contact.java [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+public class Contact {
+    String lastName;
+    String firstName;
+    String email;
+    String Title;
+
+    /**
+     * Method to return the lasName from the user.
+     *
+     * @return lastName
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/entities/Contact.java]: Please, place descriptions for returns.

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+     * @param Contactbody
+     */
+    public void setContent(String Contactbody) {
+        restClientApi.buildSpec(Contactbody);
+    }
+
+    /**
+     * Method let set task body that let build request.
+     *
+     * @param Contactbody
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/core/api/ContactApi.java]: Please, place descriptions for input parameters.

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+     * @return ContactApi.
+     */
+    public static ContactApi getInstance() {
+        return new ContactApi();
+    }
+
+    /**
+     * Method let set task body that let build request.
+     *
+     * @param Contactbody
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/core/api/ContactApi.java]: Please, place descriptions for input parameters.

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    public ContactApi() {
+        restClientApi = RestClientApi.getInstance();
+        restClientApi.setRequest(Authentication.requestSpecification());
+    }
+
+    /**
+     * Method let get the new instance to the class.
+     *
+     * @return ContactApi.
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/core/api/ContactApi.java]: Please, place descriptions for returns.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 21]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    @When("User send de request post to contact endpoint with firstName, email, title")
+    public void userSendDeRequestPostToContactEndpointWithFirstNameEmailTitle() {
+        response = RestClientApi.getInstance().post(EndPoints.CONTACT_ENDPOINT);
+    }
+
+    /**
+     * Method to let get the statutus with the first name, email and title.
+     *
+     * @param statusCode
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 22]
```diff
@@ -0,0 +1,268 @@
...
+    @When("User deletes an existing contact by Id")
+    public void userDeletesAnExistingContactById() {
+        response = requestManager.delete(EndPoints.CONTACT_ENDPOINT + "/" + "00Q3i000002MRlqEA1");
+        response.prettyPrint();
+    }
+
+    /**
+     * Method to le Delete a contact by id.
+     *
+     * @param arg0
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please use an appropriate name for this input parameter.

Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 23]
```diff
@@ -0,0 +1,268 @@
...
+    @When("User finds an existing Contact by Id")
+    public void userFindsAnExistingContactById() {
+        response = requestManager.get(EndPoints.CONTACT_ENDPOINT + "/" + "");
+        response.prettyPrint();
+    }
+
+    /**
+     * Method to led find a contact by id.
+     *
+     * @param arg0
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please use an appropriate name for this input parameter.

Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 24]
```diff
@@ -0,0 +1,268 @@
...
+    @When("User retrieves the summary for Contact")
+    public void userRetrievesTheSummaryForContact() {
+        response = requestManager.get(EndPoints.CONTACT_ENDPOINT);
+        response.prettyPrint();
+    }
+
+    /**
+     * Method to le verify status code.
+     *
+     * @param statusCode
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please use an appropriate name for this input parameter.

Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 25]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    @Then("The status code is (\\d+)")
+    public void verify_status_code(int statusCode) {
+        Assert.assertEquals(response.getStatusCode(), statusCode);
+    }
+
+    /**
+     * Method to let set build specifications with the last name.
+     *
+     * @param inputFields
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 26]
```diff
@@ -0,0 +1,268 @@
...
+    @When("User updates existing contact by Id")
+    public void userUpdatesExistingContactById() {
+        response = requestManager.patch(EndPoints.CONTACT_ENDPOINT + "/" + "");
+        response.prettyPrint();
+    }
+
+    /**
+     * Method to let set the last name to the contact.
+     *
+     * @param inputFields
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 27]
```diff
@@ -0,0 +1,268 @@
...
+     */
+    @Given("User sets lastName to the contact")
+    public void userSetsLastNameToTheContact(Map<String, String> inputFields) {
+        requestManager.buildSpec(inputFields);
+    }
+
+    /**
+     * Method to let updates contact by id.
+     *
+     * @param arg3
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please use an appropriate name for this input parameter.
Please, place descriptions for input parameters.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 28]
```diff
@@ -0,0 +1,268 @@
...
+    public void userUpdatesContactByIdFSA(int arg3) {
+        response = requestManager.get(EndPoints.CONTACT_ENDPOINT + "/" + arg3);
+        response.prettyPrint();
+
+    }
+
+    /**
+     * Method to let build specifications all right the user provider json.
+     *
+     * @param jsonBodyString
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please, place descriptions for input parameters.

File: src/test/resources/TaskAcceptance.feature [hunk 29]
```diff
@@ -0,0 +1,30 @@
...
+  Scenario: User creates a task by specifying at least a status and priority
+    Given user specifies new body content
+      | Status   | Not Started |
+      | Priority | Low         |
+    When user posts to Task endpoint
+    Then the status code is 201 after creating
+    And creation response includes the following
+      | success | true |
+
+#  Scenario: User updates a task by id and specifying at least a status and priority
```
Comment (alszla) [inline src/test/resources/TaskAcceptance.feature:15]: Remove this commented block.

File: src/test/resources/features/contact/CreateContact.feature [hunk 30]
```diff
@@ -0,0 +1,36 @@
...
+      | email      |  testemail  |
+    When User send de request post to contact endpoint with firstName, email
+    Then User get a "201" status code with firstName, email
+
+  @funtional
+  Scenario: User creates a new user contact with firstName, email, title
+    Given User specifies new contact with firstName, email, title
+      | lastName   |  testName   |
+      | firstname  |  testfirstName  |
+      | email   |  testemail  |
```
Comment (alszla) [inline src/test/resources/features/contact/CreateContact.feature]: It'd be a lot advisable if you align those "|" characters.

File: src/test/resources/features/contact/RetrieveSumaryContact.feature [hunk 31]
```diff
@@ -0,0 +1,7 @@
+Feature: Manage Tasks in Salesforce
+  Background: User is authenticated
+    Given a user sing in the web page SalesForce
+  @Acceptance
+  Scenario: Verify get summary for Contact
+    When User retrieves the summary for Contact
+    Then The status code is 200
```
Comment (alszla) [inline src/test/resources/features/contact/RetrieveSumaryContact.feature]: Add a blank line at the end of this file.

File: src/test/resources/features/contact/UpdateContact.feature [hunk 32]
```diff
@@ -0,0 +1,28 @@
+Feature: Manage Tasks in Salesforce
+  Background: User is authenticated
+    Given a user sing in the web page SalesForce
+
+@Acceptance
+Scenario: Update a Contact
+Given User sets the lastName
```
Comment (alszla) [inline src/test/resources/features/contact/UpdateContact.feature]: Given, When and Then should be indented in respect to Scenario alignment.
In similar fashion, Scenario should also be indented in respecto to Feature alignment.

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java [hunk 33]
```diff
@@ -0,0 +1,65 @@
...
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information of
+ * Jala Foundation, ("Confidential Information").  You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Jala Foundation.
+ */
+package com.foundation.salesforce.core.api;
+
```
Comment (alszla) [inline src/main/java/com/foundation/salesforce/core/api/ContactApi.java:14]: This blank line should not be here.

Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/core/api/ContactApi.java:14]: Ok. That was Done.

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java (hunk 1 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/core/api/ContactApi.java:18]: Done.

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java (hunk 20 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/core/api/ContactApi.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 2 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 3 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 4 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 5 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 34]
```diff
@@ -64,7 +64,7 @@ public void userSendDeRequestPostToContactEndpoint() {
     /**
      * Method to let compare a recuest.
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please correct the misspelling.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 35]
```diff
@@ -64,7 +64,7 @@ public void userSendDeRequestPostToContactEndpoint() {
...
      *
-     * @param statusCode
+     * @param statusCode add
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please add a more descriptive comment.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 36]
```diff
@@ -130,7 +130,7 @@ public void userGetAStatusCodeWithFirstNameEmail(String statusCode) {
     /**
      * Method to let user specifies new contact with first name, email and title.
      *
-     * @param inputContent
+     * @param inputContent
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Please add a description.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java [hunk 37]
```diff
@@ -0,0 +1,268 @@
...
+
+import java.util.Map;
+
+/**
+ * Class where it is stored the steps for the feature are carried out.
+ *
+ * @author Jesus Menacho
+ * @version 1.0
+ */
+public class ContactAcceptanceSteps {
```
Comment (alszla) [inline src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: You should move/rename this file to comply to the structure our group has adopted.
That is a single feature file for every functionality offered by the Contact endpoint.
All your feature files should also be contained within a dedicated directory.
Check the current directory structure for guidance.

File: src/main/java/com/foundation/salesforce/entities/Contact.java (hunk 11 above)
Comment (alszla) [reply src/main/java/com/foundation/salesforce/entities/Contact.java]: Many of these comments are not solved yet as they still appear unedited. Please review them.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 36 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 35 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 37 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 34 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/main/java/com/foundation/salesforce/entities/Contact.java (hunk 13 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/entities/Contact.java]: Done.

File: src/main/java/com/foundation/salesforce/entities/Contact.java (hunk 12 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/entities/Contact.java]: Done

File: src/main/java/com/foundation/salesforce/entities/Contact.java (hunk 11 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/entities/Contact.java]: Done.

File: src/main/java/com/foundation/salesforce/entities/Contact.java (hunk 10 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/entities/Contact.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 9 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 8 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 7 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 6 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/main/java/com/foundation/salesforce/entities/Contact.java (hunk 14 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/entities/Contact.java]: Done.

File: src/main/java/com/foundation/salesforce/entities/Contact.java (hunk 15 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/entities/Contact.java]: Done.

File: src/main/java/com/foundation/salesforce/entities/Contact.java (hunk 16 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/entities/Contact.java]: Done.

File: src/main/java/com/foundation/salesforce/entities/Contact.java (hunk 17 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/entities/Contact.java]: Done.

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java (hunk 18 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/core/api/ContactApi.java]: Done.

File: src/main/java/com/foundation/salesforce/core/api/ContactApi.java (hunk 19 above)
Comment (TheYisusGithub) [reply src/main/java/com/foundation/salesforce/core/api/ContactApi.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 21 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 22 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 23 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 24 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java (hunk 28 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactAcceptanceSteps.java]: Done.

File: src/test/resources/TaskAcceptance.feature (hunk 29 above)
Comment (TheYisusGithub) [reply src/test/resources/TaskAcceptance.feature:15]: Done.

File: src/test/resources/features/contact/CreateContact.feature (hunk 30 above)
Comment (TheYisusGithub) [reply src/test/resources/features/contact/CreateContact.feature]: Done.

File: src/test/resources/features/contact/RetrieveSumaryContact.feature (hunk 31 above)
Comment (TheYisusGithub) [reply src/test/resources/features/contact/RetrieveSumaryContact.feature]: Done.

**Review by alszla — APPROVED** [review]

File: src/test/java/com/foundation/salesforce/steps/ContactSteps.java [hunk 38]
```diff
@@ -0,0 +1,314 @@
...
+ * @author Jesus Menacho
+ * @version 1.0
+ */
+public class ContactSteps {
+    private ContactApi contactApi;
+    private Context context;
+
+    /**
+     * Class constructor.
+     * A class constructor instantiates a very shiny and beautiful TaskSteps object.
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/ContactSteps.java]: Please specify the name of your class.

File: src/test/java/com/foundation/salesforce/steps/ContactSteps.java [hunk 39]
```diff
@@ -0,0 +1,314 @@
...
+    }
+
+    /**
+     * Method to led a specifies a new contact.
+     *
+     * @param inputContent specified as data table in gherkin feature file.
+     */
+    @Given("User specifies new contact with firstName")
+    public void userSpecifiesNewContactWithFirstName(Map<String, String> inputContent) {
+//        RestClientApi.getInstance().buildSpec(inputContent);
```
Comment (MelissaRoman06) [inline src/test/java/com/foundation/salesforce/steps/ContactSteps.java]: Please delete this comment

File: src/test/resources/TaskAcceptance.feature [hunk 40]
```diff
@@ -0,0 +1,30 @@
...
+#    Given user specifies updated body content
+#      | Status   | Not Started |
```
Comment (MelissaRoman06) [inline src/test/resources/TaskAcceptance.feature:17]: Please delete the comments.

File: src/test/java/com/foundation/salesforce/steps/ContactSteps.java (hunk 38 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactSteps.java]: Done.

File: src/test/java/com/foundation/salesforce/steps/ContactSteps.java (hunk 39 above)
Comment (TheYisusGithub) [reply src/test/java/com/foundation/salesforce/steps/ContactSteps.java]: Done.

File: src/test/resources/TaskAcceptance.feature (hunk 40 above)
Comment (TheYisusGithub) [reply src/test/resources/TaskAcceptance.feature:17]: That is to Alejandro method.

**Review by MelissaRoman06 — APPROVED** [review]

**Pull request merged**